cpp/SnowflakeS3Client.cpp

```diff
@@ -98,6 +121,126 @@ TransferOutcome SnowflakeS3Client::upload(FileMetadata *fileMetadata,
   }
 }
 
+void *Snowflake::Client::SnowflakeS3Client::uploadParts(MultiUploadCtx * uploadCtx)
+{
+  Aws::S3::Model::UploadPartRequest uploadPartRequest;
+
+  uploadPartRequest.WithBucket(uploadCtx->m_bucket)
+                   .WithKey(uploadCtx->m_key);
+
+  uploadPartRequest.SetContentType(CONTENT_TYPE_OCTET_STREAM);
+  uploadPartRequest.SetContentLength(uploadCtx->buf->getSize());
+  uploadPartRequest.SetBody(Aws::MakeShared<Aws::IOStream>("", uploadCtx->buf));
+  uploadPartRequest.SetUploadId(uploadCtx->m_uploadId);
+  uploadPartRequest.SetPartNumber(uploadCtx->m_partNumber);
+
+  Aws::S3::Model::UploadPartOutcome outcome = s3Client->UploadPart(uploadPartRequest);
+
+  if (outcome.IsSuccess())
+  {
+    uploadCtx->m_outcome = TransferOutcome::SUCCESS;
+    uploadCtx->m_etag = outcome.GetResult().GetETag();
+  }
+  else
+  {
+    fprintf(stderr, "Upload request: %s\n", outcome.GetError().GetMessage().c_str());
```

Comment:
let's clean these message up later in the logging API in C++.

Reply:
Yeah will do that.